### PR TITLE
Qloss

### PR DESCRIPTION
**Pull request by TamerSobhy**
Hello @awig and @fxz89, 

Please check the code for reactive power loss. The attached file includes all the equations. 
[NetQloss.pdf](https://github.com/romcon/PFNET/files/4709012/NetQloss.pdf)

Looking forward to your feedback :)  

Tamer

### CODE REVIEW

**Comment by fxz89** [comment]
Hi @Tamer, the results and Qloss function seems good!

**Comment by TamerSobhy** [comment]
I will close this one as we I created PR for the updated code with subsysflag

**Comment by awig** [comment]
@TamerSobhy Sorry wasn't too clear. Let's do a PR without the the subsystem and then we can do one with the subsystem. However, I think we need to discuss more about adding the is_subsys attribute as it could apply to other types of functions and constraints as well so we should be careful with the wording and clarity for it. 

@fxz89 and I will still review this PR first.

**Comment by TamerSobhy** [comment]
@awig sure, that is better. 
The optimization of the subsystem will be different for different functions/constraints. We can discuss this latter after you approve this pull request. 

Looking forward to your feedback

**Comment by awig** [comment]
Starting to review today. Sorry its been so late.

**Comment by TamerSobhy** [comment]
No worries at all, I am available anytime if you want to discuss something.

**Comment by awig** [comment]
Thanks @TamerSobhy .  Would you please confirm you are using Windows and cmake for the build process?

**Comment by TamerSobhy** [comment]
Yes, I am using windows and Visual Studio 16 2019

File: src/problem/func/func_Q_LOSS.c [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+Func* FUNC_Q_LOSS_new(REAL weight, Net* net) {
+  Func* f = FUNC_new(weight,net);
+  FUNC_set_func_count_step(f,&FUNC_Q_LOSS_count_step);
+  FUNC_set_func_analyze_step(f,&FUNC_Q_LOSS_analyze_step);
+  FUNC_set_func_eval_step(f,&FUNC_Q_LOSS_eval_step);
+  FUNC_set_name(f,"reactive power loss");
+  return f;
+}
+
+void FUNC_Q_LOSS_count_step(Func* f, Bus* bus, BusDC* busdc, int t) {
```
Comment (awig) [inline src/problem/func/func_Q_LOSS.c:22]: i'm no DC expert, so is it safe to ignore DC buses (I think so but just wanted to see if you had any thoughts?)

Comment (TamerSobhy) [reply src/problem/func/func_Q_LOSS.c:22]: For the Q losses yes, there is no reactive power in DC system.

File: src/problem/func/func_Q_LOSS.c [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+      index_v_mag = BUS_get_index_v_mag(bus,t);
+      v = VEC_get(var_values,index_v_mag);
+  }
+  else {
+      v = BUS_get_v_mag(bus,t);
+  }
+
+  // Branches_k
+  for (br = BUS_get_branch_k(bus); br != NULL; br = BRANCH_get_next_k(br)) {
```
Comment (awig) [inline src/problem/func/func_Q_LOSS.c:214]: Any chance we are double counting branch susceptances twice from by using branch_k and branch_m? I don't think so but just wanted to see if you see any potential for it.

Comment (TamerSobhy) [reply src/problem/func/func_Q_LOSS.c:214]: We are counting from one side (k-side), which will ensure the branches counted once. (The same concept also was used before to calculate the interface losses in GAT/functions) So don't worry about it.

**Comment by awig** [comment]
Looks great! Thanks @TamerSobhy 

I just made some small changes to the comments and merged it with dev plus a few changes that were needed to get it building correctly with autotools.

**Pull request merged**